openmetadata-ui/src/main/resources/ui/src/pages/GlossaryPage/GlossaryPageV1.component.tsx

```diff
@@ -209,10 +189,129 @@ const GlossaryPageV1 = () => {
+            const total = res.data.hits.total.value;
+
+            return forceReset
+              ? {
+                  data,
+                  total,
+                  currPage: 1,
+                }
+              : { ...pre, data, total };
+          });
+        } else {
+          setAssetData((pre) => {
+            const data = [] as GlossaryTermAssets['data'];
+            const total = 0;
+
+            return forceReset
+              ? {
+                  data,
+                  total,
+                  currPage: 1,
+                }
+              : { ...pre, data, total };
+          });
+        }
+      });
+    } else {
+      setAssetData({ data: [], total: 0, currPage: 1 });
+    }
+  };
+
+  const selectDataByFQN = (
+    dataFQN: string,
+    arrGlossary: ModifiedGlossaryData[]
+  ) => {
+    handleChildLoading(true);
+    const hierarchy = getTermPosFromGlossaries(arrGlossary, dataFQN);
+    if (hierarchy.length < 2) {
+      setSelectedData(arrGlossary[hierarchy[0]]);
+      handleSelectedKey(dataFQN);
+      if (!expandedKey.length) {
+        setExpandedKey([dataFQN]);
+      }
+      setIsGlossaryActive(true);
+      setIsLoading(false);
+      handleChildLoading(false);
+    } else {
+      setLoadingKey((pre) => {
+        return !pre.includes(dataFQN) ? [...pre, dataFQN] : pre;
+      });
+      fetchGlossaryTermByName(dataFQN, hierarchy, arrGlossary);
+      fetchGlossaryTermAssets(dataFQN, true);
+    }
+  };
+
+  const checkAndFetchDataByFQN = (
+    arrGlossary: ModifiedGlossaryData[],
+    fqn: string
+  ) => {
+    let modifiedData = cloneDeep(arrGlossary);
+    const arrFQN = getHierarchicalKeysByFQN(fqn);
+    const glossary: ModifiedGlossaryData | GlossaryTerm = modifiedData.find(
+      (item) => item.name === arrFQN[0]
+    ) as ModifiedGlossaryData;
+    const data = getTermDataFromGlossary(glossary, fqn);
+    if (isEmpty(data)) {
+      modifiedData = updateGlossaryListBySearchedTerms(modifiedData, [
+        { fqdn: arrFQN[arrFQN.length - 1] },
+      ] as FormattedGlossarySuggestion[]);
+    }
+    selectDataByFQN(fqn, modifiedData);
+  };
+
+  const fetchGlossaryList = (termFqn = '', paging = '') => {
+    setIsLoading(true);
+    getGlossariesWithRootTerms(paging, 1000, ['owner', 'tags', 'reviewers'])
+      .then((data: ModifiedGlossaryData[]) => {
+        if (data?.length) {
+          setGlossaries(data);
+          setGlossariesList(data);
+          if (termFqn) {
+            checkAndFetchDataByFQN(data, termFqn);
+          } else {
+            initSelectGlossary(data[0]);
+            setIsLoading(false);
+            handleChildLoading(false);
+          }
+        } else {
+          setGlossariesList([]);
+          setIsLoading(false);
+          handleChildLoading(false);
+        }
+      })
+      .catch((err: AxiosError) => {
+        showToast({
+          variant: 'error',
+          body: err.response?.data?.message ?? 'Something went wrong!',
```

Comment:
We should let the user know what went wrong.

Reply:
Done